sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -727,9 +716,16 @@ object ViewSyncSchemaToMetaStore extends (LogicalPlan => Unit) {
           }
           SchemaUtils.checkColumnNameDuplication(fieldNames.toImmutableArraySeq,
             session.sessionState.conf.resolver)
-          val updatedViewMeta = metaData.copy(
-            properties = newProperties,
-            schema = newSchema)
+          val updatedViewMeta =
+            if (viewSchemaMode == SchemaEvolution) {
+              metaData.copy(
```

Comment:
Note:  we already seem to use the ViewSchema directly in the table resolution for SchemaEvolution mode:  https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L991  iiuc